tslearn/docs/user_guide/matrix-profile.rst

```diff
@@ -0,0 +1,37 @@
+.. _matrix-profile:
+
+Matrix Profile
+==============
+
+The Matrix Profile, :math:`MP`, is a new time series that can be calculated based on an input time series :math:`T` and a subsequence length :math:`m`. :math:`MP_i` corresponds to the minimal distance from the query subsequence :math:`T_{i\rightarrow i+m}` to :math:`T` [1]_. The region around query subsequence is not included when calculating the minimal distances, as these trivial matches would result in distances of :math:`0`. The Matrix Profile is closely related to the distance calculation used to transform time series into their shapelet-transform space, as illustrated below.
+
+.. figure:: ../_static/img/distance_matrix_profile.svg
+    :width: 80%
+    :align: center
```

Comment:
Maybe it would be nice that the user guide takes more time to detail the extraction scheme. 

I used [this tutorial](https://stumpy.readthedocs.io/en/latest/Tutorial_The_Matrix_Profile.html) as a guide for my implementation. It's probably too detailed, but it really helps understanding what the MP transform is.

Reply:
What do you exactly mean with the extraction scheme? How segments are extracted and how these are again normalized?

Reply:
Something like they have in their stumpy tutorial, but will fewer details: a step-by-step example that explains the computations involved so that someone that is not familiar with MP can quickly get the underlying idea of how it's computed.

Reply:
Ok! I tried to achieve that with the current figure already, but I can definitely extend it. I'll take a look at the tutorial.

Reply:
I just added another small paragraph to refer to StumPy (if users are only requiring a matrix profile implementation, they better go there). Perhaps it's kind of silly to re-make their tutorial as well? It is a pretty impressive one with the animations etc!

Reply:
Thank you all for the positive notes! `tslearn` is great for the community!

@rtavenar Just a word of warning that [the tutorial](https://stumpy.readthedocs.io/en/latest/Tutorial_The_Matrix_Profile.html) was meant to be rather naive and that, for the sake of clarity, it purposely omits z-normalization of the subsequences (it only computes straight Euclidean distances which is wrong) and so STUMPY's matrix profile calculations actually do a little bit more and leverages algorithmic optimizations that make the computations O(n*n) complexity (rather than O(m*n*n)).

Reply:
Yep, we do the normalization, but we don't use the trick.

Reply:
Great. So, depending on the length of your time series (and your implementation), you'll start noticing a "slow down" around length 1,000-5,000